force-app/main/interoperability/classes/ProductRegistryMapper.cls

```diff
@@ -0,0 +1,94 @@
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+ * @author Salesforce.org
+ * @date 2021
+ * @group
+ * @group-content
+ * @description Mapper class to transform Product_Registry__mdt into ProductRegistryModel
+ */
+public virtual with sharing class ProductRegistryMapper {
+    public static final string PRODUCT_INFORMATION_ACTION = 'Settings Product Information';
+
+    @TestVisible
+    private static ProductRegistryMapper instance;
+
+    /*********************************************************************
+     * @description Empty constructor is not public for Singleton Pattern
+     **********************************************************************/
+    protected ProductRegistryMapper() {
+    }
+
+    /*********************************************
+     * @description Instance for Singleton Pattern
+     **********************************************/
+    public static ProductRegistryMapper getInstance() {
+        if (instance == null) {
+            instance = new ProductRegistryMapper();
+        }
+
+        return instance;
+    }
+    /********************************************************************************************
+     * @description Returns the list of ProductRegistryModel based on the Product Registries loaded by DB
+     * @return   A list of ProductRegistryModel with all its properties populated
+     ********************************************************************************************/
+    public virtual List<ProductRegistryModel> getProductRegistrySettingsProductInformation() {
+        List<ProductRegistryModel> productRegistryModelList = new List<ProductRegistryModel>();
+        List<Product_Registry__mdt> productRegistrySettingsProductInformationList = this.getProductRegistriesByAction(
+            PRODUCT_INFORMATION_ACTION
+        );
+
+        if (productRegistrySettingsProductInformationList != null) {
+            for (Product_Registry__mdt productRegistry : productRegistrySettingsProductInformationList) {
+                ProductRegistryModel prodRegistryModel = new ProductRegistryModel(
+                    productRegistry.Action__c,
+                    productRegistry.Namespace__c,
+                    productRegistry.Class_Name__c,
+                    productRegistry.API_Version__c
+                );
+                productRegistryModelList.add(prodRegistryModel);
+            }
+        }
```

Comment:
This structure to the code was hiding that we didn't have a unit test for productRegistrySettingsProductInformationList == null as well as if that is a testable use case. Does an empty query instead result in an empty list of Product_Registry__mdt? We may not even need to test for that situation.

```suggestion
        if (productRegistrySettingsProductInformationList == null) {
            return null;
        }

        for (Product_Registry__mdt productRegistry : productRegistrySettingsProductInformationList) {
            ProductRegistryModel prodRegistryModel = new ProductRegistryModel(
                productRegistry.Action__c,
                productRegistry.Namespace__c,
                productRegistry.Class_Name__c,
                productRegistry.API_Version__c
            );
            productRegistryModelList.add(prodRegistryModel);
        }
```